providers/tailscale/provider/provider.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+package provider
+
+import (
+	"context"
+	"errors"
+
+	"go.mondoo.com/cnquery/v11/llx"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/upstream"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/vault"
+	"go.mondoo.com/cnquery/v11/providers/tailscale/connection"
+	"go.mondoo.com/cnquery/v11/providers/tailscale/resources"
+)
+
+const DefaultConnectionType = "tailscale"
+
+type Service struct {
+	*plugin.Service
+}
+
+func Init() *Service {
+	return &Service{
+		Service: plugin.NewService(),
+	}
+}
+
+func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error) {
```

Comment:
Using an empty token does not throw an error in the initial phase:

```
echo $TAILSCALE_API_KEY

cnquery shell tailscale --token $TAILSCALE_API_KEY --discover devices
→ tailscale> authentication configured method=token
→ connected to Tailscale
  ___ _ __   __ _ _   _  ___ _ __ _   _ 
 / __| '_ \ / _` | | | |/ _ \ '__| | | |
| (__| | | | (_| | |_| |  __/ |  | |_| |
 \___|_| |_|\__, |\__,_|\___|_|   \__, |
  mondoo™      |_|                |___/  interactive shell

cnquery> tailscale
tailscale: tailscale id = devices
cnquery> tailscale.devices
API token invalid (401)
tailscale.devices: no data available
```

Reply:
I will add a verify step to fail fast, just like we do with the GitHub provider, good call! 💯